src/apigateway/commands/copyUrl.ts

```diff
@@ -48,7 +48,7 @@ export async function copyUrlCommand(
         )
     } catch (e) {
         getLogger().error(`Failed to load stages: %O`, e)
-        recordApigatewayCopyUrl({ result: 'Failed' })
+        telemetry.apigateway_copyUrl.emit({ result: 'Failed' })
```

Comment:
The `emit` method is equivalent to the old behavior. I chose the verb `emit` because `record` now refers to placing data into the span but not necessarily sending it.

Reply:
Does `record`  need to ever be explicitly called for the metric to be posted? What happens if `emit()` is never followed bye `record()` ?

Reply:
> Does record need to ever be explicitly called for the metric to be posted? 

No. The closest analogy here is comparing `record` to a log statement. `getLogger().error` doesn't get to choose exactly when (or what) is written to the log file, it only outputs some basic information that may eventually be written to the logs. `emit` is kind of like a 'flush' in that it immediately sends the contents of the span to telemetry.

But just like _most_ developers don't care when the logs flush, _most_ developers won't need to worry about `emit`. 

> What happens if emit() is never followed bye record() ?

If the span was creating outside of `run`, nothing would be sent. A span would exist until the execution frame is dropped, then it gets garbage collected. There's several ways to handle this scenario but I haven't decided about what makes the most sense.